products/compute/api.yaml

```diff
@@ -15553,6 +15870,496 @@ objects:
+                  with contents of hostRewrite.
+
+                  The value must be between 1 and 255 characters.
+                at_least_one_of:
+                  - default_route_action.0.url_rewrite.0.path_prefix_rewrite
+                  - default_route_action.0.url_rewrite.0.host_rewrite
+          - !ruby/object:Api::Type::NestedObject
+            name: 'timeout'
+            description: |
+              Specifies the timeout for the selected route. Timeout is computed from the time the request has been
+              fully processed (i.e. end-of-stream) up until the response has been completely processed. Timeout includes all retries.
+
+              If not specified, will use the largest timeout among all backend services associated with the route.
+            at_least_one_of:
+              - default_route_action.0.weighted_backend_services
+              - default_route_action.0.url_rewrite
+              - default_route_action.0.timeout
+              - default_route_action.0.retry_policy
+              - default_route_action.0.request_mirror_policy
+              - default_route_action.0.cors_policy
+              - default_route_action.0.fault_injection_policy
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'seconds'
+                description: |
+                  Span of time at a resolution of a second. Must be from 0 to 315,576,000,000 inclusive.
+                  Note: these bounds are computed from: 60 sec/min * 60 min/hr * 24 hr/day * 365.25 days/year * 10000 years
+                at_least_one_of:
+                  - default_route_action.0.timeout.0.seconds
+                  - default_route_action.0.timeout.0.nanos
+              - !ruby/object:Api::Type::Integer
+                name: 'nanos'
+                description: |
+                  Span of time that's a fraction of a second at nanosecond resolution. Durations less than one second are represented
+                  with a 0 seconds field and a positive nanos field. Must be from 0 to 999,999,999 inclusive.
+                at_least_one_of:
+                  - default_route_action.0.timeout.0.seconds
+                  - default_route_action.0.timeout.0.nanos
+          - !ruby/object:Api::Type::NestedObject
+            name: 'retryPolicy'
+            description: |
+              Specifies the retry policy associated with this route.
+            at_least_one_of:
+              - default_route_action.0.weighted_backend_services
+              - default_route_action.0.url_rewrite
+              - default_route_action.0.timeout
+              - default_route_action.0.retry_policy
+              - default_route_action.0.request_mirror_policy
+              - default_route_action.0.cors_policy
+              - default_route_action.0.fault_injection_policy
+            properties:
+              - !ruby/object:Api::Type::Array
+                name: 'retryConditions'
```

Comment:
does this field need to be a set?

Reply:
i think it should be fine as an array - i ran a few tests on it and it seemed to be fine, and also the same `retryConditions` is on `pathMatchers.routeRules.routeAction.retryPolicy.retryConditions` as an array, so I feel pretty comfortable that it's been working fine.